src/MagicLogin.php

```diff
@@ -113,6 +115,58 @@ function (RegisterTemplateRootsEvent $event) {
             }
         );
 
+        Event::on(
+            UsersController::class,
+            UsersController::EVENT_BEFORE_ACTION,
+            function (ActionEvent $event) {
+                if (!$event->sender->action->actionMethod == 'actionSaveUser') {
+                    return;
+                }
+
+                // If we are updating an existing user then skip this.
+                $userId = $this->request->getBodyParam('userId');
+                if ($userId) {
+                    return;
+                }
+
+                $event->sender->requirePostRequest();
+                
+                // Require email.
+                $email = $this->request->getRequiredBodyParam('email');
+                if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+                    // TODO: Set this to be configurable.
+                    $event->sender->setFailFlash(\Craft::t('magic-login', 'Please enter a valid email address.'));
+                    $event->isValid = false;
+                    return;
+                }
+
+                // TODO: What do we do if already registered. Do we throw an error?
```

Comment:
This could just be left to the controller action to figure it out.

Reply:
Actually we should leave this in. I'd like more control over the error thrown if this does happen.